fMRIVolume/scripts/IntensityNormalization.sh

```diff
@@ -117,20 +140,49 @@ echo "PWD = `pwd`" >> $WD/log.txt
-
-# FinalMask is a combination of the FS-derived brainmask, and the spatial coverage mask that captures the
-# voxels that have data available at *ALL* time points (derived in OneStepResampling)
 FinalMask=`${FSLDIR}/bin/remove_ext ${InputfMRI}`_finalmask
-${FSLDIR}/bin/fslmaths ${BrainMask} -bin -mas ${InputfMRI}_mask ${FinalMask}
+T1FOVMask=`${FSLDIR}/bin/remove_ext ${InputfMRI}`_T1FOVmask
+${FSLDIR}/bin/fslmaths ${BrainMask} -bin -mas ${InputfMRI}_mask ${T1FOVMask}
+
+# Use the requested BOLD mask 
+
+if [ "${BOLDMask}" = "T1_fMRI_FOV" ] ; then
+    # FinalMask is a combination of the FS-derived brainmask, and the spatial coverage mask that captures the
+    # voxels that have data available at *ALL* time points (derived in OneStepResampling)
+    ${FSLDIR}/bin/imcp ${T1FOVMask} ${FinalMask}
+
+elif [ "${BOLDMask}" = "T1_DILATED_fMRI_FOV" ] ; then
+    # FinalMask is a combination of once dilated FS-derived brainmask, and the spatial coverage mask that captures the
+    # voxels that have data available at *ALL* time points (derived in OneStepResampling)
+    ${FSLDIR}/bin/fslmaths ${BrainMask} -bin -dilF -mas ${InputfMRI}_mask ${FinalMask}
+
+elif [ "${BOLDMask}" = "T1_DILATED2x_fMRI_FOV" ] ; then
+    # FinalMask is a combination of twice dilated FS-derived brainmask, and the spatial coverage mask that captures the
+    # voxels that have data available at *ALL* time points (derived in OneStepResampling)
+    ${FSLDIR}/bin/fslmaths ${BrainMask} -bin -dilF -dilF -mas ${InputfMRI}_mask ${FinalMask}
+
+elif [ "${BOLDMask}" = "fMRI_FOV" ] ; then
+    # FinalMask is the spatial coverage mask that captures the
+    # voxels that have data available at *ALL* time points (derived in OneStepResampling)
+    ${FSLDIR}/bin/imcp ${InputfMRI}_mask ${FinalMask}
+
+elif
+    # No valid BOLDMask option was specified
+    log_Err_Abort "Invalid entry for specified BOLD mask (--boldmask=${BOLDMask})"
+fi
 
-# Create a simple summary text file of the percentage of spatial coverage of the fMRI data inside the FS-derived brain mask
+
+# Create a simple summary text file of the percentage of spatial coverage of the fMRI data inside the FS-derived brain mask and the actual mask used
 NvoxBrainMask=`fslstats ${BrainMask} -V | awk '{print $1}'`
+NvoxT1FOVMask=`fslstats ${T1FOVMask} -V | awk '{print $1}'`
 NvoxFinalMask=`fslstats ${FinalMask} -V | awk '{print $1}'`
-PctCoverage=`echo "scale=4; 100 * ${NvoxFinalMask} / ${NvoxBrainMask}" | bc -l`
-echo "PctCoverage, NvoxFinalMask, NvoxBrainMask" >| ${FinalMask}.stats.txt
-echo "${PctCoverage}, ${NvoxFinalMask}, ${NvoxBrainMask}" >> ${FinalMask}.stats.txt
+PctBrainCoverage=`echo "scale=4; 100 * ${NvoxT1FOVMask} / ${NvoxBrainMask}" | bc -l`
+PctMaskCoverage=`echo "scale=4; 100 * ${NvoxFinalMask} / ${NvoxT1FOVMask}" | bc -l`
+echo "Mask, PctBrainCoverage, PctMaskCoverage, NvoxT1FOVMask, NvoxBrainMask, NvoxFinalMask" >| ${FinalMask}.stats.txt
```

Comment:
Maybe just call the first column BOLDMask to tie back to the argument name.

Reply:
Done!